src/fix_bd_sphere.cpp

```diff
@@ -0,0 +1,303 @@
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Originally modified from USER-CGDNA/fix_nve_dotc_langevin.cpp. 
+
+   Contributing author: Sam Cameron (University of Bristol)
+------------------------------------------------------------------------- */
+
+#include "fix_bd_sphere.h"
+
+#include <cmath>
+#include <cstring>
+#include "math_extra.h"
+#include "atom.h"
+#include "force.h"
+#include "update.h"
+#include "comm.h"
+#include "domain.h"
+#include "random_mars.h"
+#include "memory.h"
+#include "error.h"
+
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+/* ---------------------------------------------------------------------- */
+
+FixBdSphere::FixBdSphere(LAMMPS *lmp, int narg, char **arg) :
+  Fix(lmp, narg, arg)
+{
+  virial_flag = 1;
+
+  time_integrate = 1;
+  
+  if (narg != 8 && narg != 10)
+    error->all(FLERR,"Illegal fix bd/sphere command.");
+
+  if (!atom->sphere_flag)
+    error->all(FLERR,"Fix bd/sphere requires atom style sphere");
+  if (!atom->mu_flag)
+    error->all(FLERR,"Fix bd/sphere requires atom attribute mu");
+
+  gamma_t = utils::numeric(FLERR,arg[3],false,lmp);
+  if (gamma_t <= 0.0)
+    error->all(FLERR,"Fix bd/sphere translational viscous drag "
+	       "coefficient must be > 0.");
+
+  gamma_r = utils::numeric(FLERR,arg[4],false,lmp);
+  if (gamma_t <= 0.0)
+    error->all(FLERR,"Fix bd/sphere rotational viscous drag "
+	       "coefficient must be > 0.");
+
+  // note that diffusion is in units of epsilon**2*tau**3/(m**2*sigma**2)
+  diff_t = utils::numeric(FLERR,arg[5],false,lmp);
+  if (diff_t <= 0.0)
+    error->all(FLERR,"Fix bd/sphere translational diffusion "
+	       "coefficient must be > 0.");
+  
+  diff_r = utils::numeric(FLERR,arg[6],false,lmp);
+  if (diff_r <= 0.0)
+    error->all(FLERR,"Fix bd/sphere rotational diffusion "
+	       "coefficient must be > 0.");
+  
+  seed = utils::inumeric(FLERR,arg[7],false,lmp);
+  if (seed <= 0) error->all(FLERR,"Fix bd/sphere seed must be > 0.");
+
+  noise_flag = 1;
+  gaussian_noise_flag = 0;
+
+  int iarg = 8;
+
+  while (iarg < narg) {
+    if (strcmp(arg[iarg],"rng") == 0) {
+      if (strcmp(arg[iarg + 1],"uniform") == 0) {
+	noise_flag = 1;
+      } else if (strcmp(arg[iarg + 1],"gaussian") == 0) {
+	noise_flag = 1;
+	gaussian_noise_flag = 1;
+      } else if (strcmp(arg[iarg + 1],"none") == 0) {
+	noise_flag = 0;
+      } else {
+	error->all(FLERR,"Illegal fix/bd/sphere command.");
+      }
+    } else {
+      error->all(FLERR,"Illegal fix/bd/sphere command.");
+    }
+    iarg = iarg + 2;
+  }
+
```

Comment:
please reduce excess empty lines for better readability of the source code

Reply:
Will do.